nltk/metrics/distance.py

```diff
@@ -103,6 +103,76 @@ def edit_distance(s1, s2, substitution_cost=1, transpositions=False):
     return lev[len1][len2]
 
 
+def _edit_dist_backtrace(lev_map):
+    i, j = len(lev_map) - 1, len(lev_map[1]) - 1
+    alignment = [(i, j)]
+
+    while (i, j) != (0, 0):
+        # find direction of minimum cost
+        s1_skip_cost = s2_skip_cost = sub_cost = float('inf')
```

Comment:
May I suggest a more declarative take on the inside of the while-loop? It takes advantage of Python's built-in `min` function and the `operator` module.

```python
directions = [
    (i - 1, j),      # skip s1
    (i, j - 1),      # skip s2
    (i - 1, j - 1),  # substitution
]
direction_costs = (
   (lev_map[i, j], (i, j)) for i, j in directions
)
_, (i, j) = min(direction_costs, key=operator.itemgetter(0))
```

Feel free to ask if anything in the snippet above is unclear!

Reply:
I've made this change.